slice expression

### The following program `sample.go` triggers an unexpected result

```go
package main

import (    
    "crypto/sha1"
    "encoding/hex"                                                                                    
)                    
                                      
func main() {                           
    script := "hello"                                                                                 
    sumRaw := sha1.Sum([]byte(script))                                                                
    sum := hex.EncodeToString(sumRaw[:])
    println(sum)
}
```


### Expected result

```console
$ go run ./sample.go
aaf4c61ddcc5e8a2dabede0f3b482cd9aea9434d
```


### Got

```console
yaegi ./sample.go 
run: ./sample.go:11:28: cannot use type [20]uint8 as type []uint8
```


### Yaegi Version

v0.11.1

### Additional Notes

_No response_

--- a/_test/issue-1328.go
+++ b/_test/issue-1328.go
@@ -1,0 +1,16 @@
+package main
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+)
+
+func main() {
+	script := "hello"
+	sumRaw := sha1.Sum([]byte(script))
+	sum := hex.EncodeToString(sumRaw[:])
+	println(sum)
+}
+
+// Output:
+// aaf4c61ddcc5e8a2dabede0f3b482cd9aea9434d

--- a/interp/type.go
+++ b/interp/type.go
@@ -3,7 +3,6 @@ package interp
 import (
 	"fmt"
 	"go/constant"
-	"log"
 	"path/filepath"
 	"reflect"
 	"strconv"
@@ -859,7 +858,6 @@ func nodeType2(interp *Interpreter, sc *scope, n *node, seen []*node) (t *itype,
 		case ptrT:
 			t = t.val
 		case valueT:
-			log.Println(n.cfgErrorf("nodeType sliceExpr"), t.rtype, t.rtype.Kind())
 			t = valueTOf(reflect.SliceOf(t.rtype.Elem()), withScope(sc))
 		}
 		if t.cat == arrayT {

--- a/interp/type.go
+++ b/interp/type.go
@@ -860,6 +860,7 @@ func nodeType2(interp *Interpreter, sc *scope, n *node, seen []*node) (t *itype,
 			case reflect.Array, reflect.Ptr:
 				t = valueTOf(reflect.SliceOf(t.rtype.Elem()), withScope(sc))
 			}
+			break
 		}
 		if t.cat == ptrT {
 			t = t.val

--- a/interp/type.go
+++ b/interp/type.go
@@ -854,13 +854,15 @@ func nodeType2(interp *Interpreter, sc *scope, n *node, seen []*node) (t *itype,
 		if err != nil {
 			return nil, err
 		}
-		if t.cat == ptrT {
+		switch t.cat {
+		case ptrT:
 			t = t.val
-		}
-		if t.cat == arrayT {
+		case arrayT:
 			incomplete := t.incomplete
 			t = sliceOf(t.val, withNode(n), withScope(sc))
 			t.incomplete = incomplete
+		case valueT:
+			t = valueTOf(reflect.SliceOf(t.rtype.Elem()), withScope(sc))
 		}
 
 	case structType:

--- a/interp/type.go
+++ b/interp/type.go
@@ -858,7 +858,10 @@ func nodeType2(interp *Interpreter, sc *scope, n *node, seen []*node) (t *itype,
 		case ptrT:
 			t = t.val
 		case valueT:
-			t = valueTOf(reflect.SliceOf(t.rtype.Elem()), withScope(sc))
+			switch t.rtype.Kind() {
+			case reflect.Array, reflect.Ptr:
+				t = valueTOf(reflect.SliceOf(t.rtype.Elem()), withScope(sc))
+			}
 		}
 		if t.cat == arrayT {
 			incomplete := t.incomplete

--- a/interp/type.go
+++ b/interp/type.go
@@ -854,14 +854,15 @@ func nodeType2(interp *Interpreter, sc *scope, n *node, seen []*node) (t *itype,
 		if err != nil {
 			return nil, err
 		}
-		switch t.cat {
-		case ptrT:
-			t = t.val
-		case valueT:
+
+		if t.cat == valueT {
 			switch t.rtype.Kind() {
 			case reflect.Array, reflect.Ptr:
 				t = valueTOf(reflect.SliceOf(t.rtype.Elem()), withScope(sc))
 			}
+		}
+		if t.cat == ptrT {
+			t = t.val
 		}
 		if t.cat == arrayT {
 			incomplete := t.incomplete

--- a/interp/type.go
+++ b/interp/type.go
@@ -3,6 +3,7 @@ package interp
 import (
 	"fmt"
 	"go/constant"
+	"log"
 	"path/filepath"
 	"reflect"
 	"strconv"
@@ -857,12 +858,14 @@ func nodeType2(interp *Interpreter, sc *scope, n *node, seen []*node) (t *itype,
 		switch t.cat {
 		case ptrT:
 			t = t.val
-		case arrayT:
+		case valueT:
+			log.Println(n.cfgErrorf("nodeType sliceExpr"), t.rtype, t.rtype.Kind())
+			t = valueTOf(reflect.SliceOf(t.rtype.Elem()), withScope(sc))
+		}
+		if t.cat == arrayT {
 			incomplete := t.incomplete
 			t = sliceOf(t.val, withNode(n), withScope(sc))
 			t.incomplete = incomplete
-		case valueT:
-			t = valueTOf(reflect.SliceOf(t.rtype.Elem()), withScope(sc))
 		}
 
 	case structType:

